src/test/java/org/opensearch/neuralsearch/e2e/HybridCollapseIT.java

```diff
@@ -67,7 +82,7 @@ public void testCollapse_whenE2E_thenSuccessful() {
         assertTrue(isCollapseDuplicateRemoved(searchResponse.toString(), collapseDuplicate));
     }
 
-    public void testCollapse_whenE2E_andSortEnabled_thenSuccessful() {
+    private void testCollapse_whenE2E_andSortEnabled_thenSuccessful() {
```

Comment:
why we are changing this?

Reply:
If they are public then they will also run as tests. I am already calling these method from the common method here https://github.com/opensearch-project/neural-search/pull/1647/files#diff-49f453d9424e2fbde9ba4f3bf555f8b7d71c0512ce6ee0359a903737fb038928R47. As it saves memory by not creating the same index and ingest same documents thrice.

Reply:
But if we are doing the exact same index with exact same documents three times earlier, and combine them in caller method reduces the task from 3 to 1.

Yes @heemin32 one shard case is needed because in Hybrid query the process of search is different in one shard when compared to multiple shard. The normalization process at the end in single shard scenario

Reply:
@heemin32 I think the current way is much better as we are achieving the same goal. Also if use @BeforeClass then we have use @AfterClass as well for cleanup.

Reply:
@martin-gaievski wdyt?

Reply:
I see pros and cons for each of the approaches, for a single index per test class main concern is - index state can be changes inside the method. Because we are not changing index content in this particular test  I'm inclining towards a single creation and deletion of the index with `before` and `after` annotations respectively. We can create/delete both indices (with 1 and 5 shards).